### PR TITLE
5279 update state handling pinia

### PR DESCRIPTION
**Pull request by weskubo-cgi**
Updated state management from vuex to pinia. Pinia is much simpler and the store is access like normal code/components using getters (computed) and actions (methods). This was a major refactor but should result in simpler code with no functional differenc.es

### CODE REVIEW

File: frontend/src/router/index.js [hunk 1]
```diff
@@ -308,13 +313,13 @@ export const createRouter = (app) => {
 
     // Secured pages shouldn't be available to unauthenticated users
     if (to.meta.requiresAuth && !authenticated) {
-      store.commit('alert/setErrorAlert', `You are not authorized to access ${to.path}. Please login first.`)
+      //store.commit('alert/setErrorAlert', `You are not authorized to access ${to.path}. Please login first.`)
```
Comment (daveb-hni) [inline frontend/src/router/index.js:316]: Is this missing the replacement pinia setting of the alert? alertStore.setErrorAlert(). Can remove commented line.

File: frontend/src/router/index.js [hunk 2]
```diff
@@ -323,11 +328,12 @@ export const createRouter = (app) => {
     // Validate routes secured by permission
     const permission = to.meta.permission
     if (permission) {
-      const hasPermission = store.getters['auth/hasPermission'](permission)
+      const hasPermission = authStore.hasPermission(permission)
       if (hasPermission) {
         next()
       } else {
-        store.commit('alert/setErrorAlert', `You are not authorized to access ${to.path}`)
+        console.log('why here?')
```
Comment (daveb-hni) [inline frontend/src/router/index.js:335]: Can remove logging if no longer needed.

**Review by daveb-hni — COMMENTED** [review]
Please see comments.

File: frontend/src/router/index.js (hunk 2 above)
Comment (weskubo-cgi) [reply frontend/src/router/index.js:335]: Clearly lol.

File: frontend/src/router/index.js (hunk 1 above)
Comment (weskubo-cgi) [reply frontend/src/router/index.js:316]: Yes.

File: frontend/src/main.js [hunk 3]
```diff
@@ -49,9 +53,13 @@ function initApp() {
   app.component('AppOutput', AppOutput)
   app.component('AppSelect', AppSelect)
 
+  // Set up Pinia with permissions
+  app.use(createPinia())
+  const auth = useAuthStore()
+  auth.permissions = permissions
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/main.js:59]: Pinia allows direct state access and doesn't deal with the concept of setters. Store has been updated.

**Review by daveb-hni — APPROVED** [review]
Latest changes look good

**Pull request merged**